### PR TITLE
aws-iam documentation

### PR DESCRIPTION
**Pull request by hyperbolic2346**
Just wanted to get this up here so you can start changing my z's to s's.

### CODE REVIEW

File: pages/k8s/aws-iam-auth.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+---
+wrapper_template: "base_docs.html"
+markdown_includes:
+  nav: "shared/_side-navigation.md"
+context:
+  title: "AWS-IAM on Charmed Kubernetes"
+  description: Using AWS credentials to authenticate and authorize on Charmed Kubernetes
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: authorise

File: pages/k8s/aws-iam-auth.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+sidebar: k8smain-sidebar
+permalink: aws-iam-auth.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+## AWS IAM
+
+[AWS IAM](https://aws.amazon.com/iam/) credentials can be used for
+authentication and authorization on your Charmed Kubernetes cluster without
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: authorisation

File: pages/k8s/aws-iam-auth.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+regard to where it is hosted. The only requirement is that the client
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: ... that both the client...

File: pages/k8s/aws-iam-auth.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+```bash
+juju deploy charmed-kubernetes  --overlay ~/path/aws-iam-overlay.yaml
+```
+
+### User Configuration
+
+The [aws-iam-authenticator](aws-iam-authenticator-github) is configured via
+[Custom Resource Definition or CRD](k8s-crd-docs)s. These resource definitions map an AWS IAM role or user
+to a [Kubernetes RBAC](k8s-rbac-docs) user or group. This means that
+authentication happens via AWS IAM credentials, but authorization depends
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: authorisation

File: pages/k8s/aws-iam-auth.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+on standard Kubernetes RBAC rules. The CRD for this mapping is called an
+IAMIdentityMapping and looks something like this:
+```yaml
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: insert a blank line before code blocks

File: pages/k8s/aws-iam-auth.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+The aws-iam-authenticator binary needs to be installed on the machine that is
+running kubectl. This is executed by kubectl in order to log in and get a token,
+which is then passed to the Kubernetes API server. You can find the binary
+on the (aws-iam-authenticator releases page)[aws-iam-authenticator-releases].
+
+#### Updating kubectl config
+
+In order to use the [aws-iam-authenticator](aws-iam-authenticator-github) with
+kubectl, an updated config file is needed. The config file written to the
+kubernetes-master unit will have a user named aws-iam-user that uses the
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: I would put backticks around aws-iam-user and aws-iam-authenticator

File: pages/k8s/aws-iam-auth.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+aws-iam-authenticator client binary and a context named aws-iam-authenticator.
+First, copy the config:
+
+```bash
+juju scp kubernetes-master/0:config ~/.kube/config
+```
+
+The config file will need to be edited in order to add the desired arn
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: I would spell out  - Amazon Resource Name(ARN)

File: pages/k8s/aws-iam-auth.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+for authentication. Information about this can be found on the
+[aws-iam-authenticator documentation](aws-iam-authenticator-config).
+
+The context that uses aws-iam-authenticator can be selected with:
+
+```bash
+kubectl config use-context aws-iam-authenticator
+```
+
+### A note about authorization
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: s/z/s

File: pages/k8s/aws-iam-auth.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+The AWS-IAM charm can be used for authentication only or can be used in an
+RBAC-enabled cluster to authorize users as well. If the charm is related to
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: authorise

File: pages/k8s/aws-iam-auth.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+a Charmed Kubernetes cluster without RBAC enabled, any valid AWS IAM
+credential that can assume a role specified in the IAMIdentityMapping
+CRD will be able to do any commands against the cluster. If RBAC is enabled,
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: ...able to run commands

File: pages/k8s/aws-iam-auth.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+authentication and authorization on your Charmed Kubernetes cluster without
+regard to where it is hosted. The only requirement is that the client
+machine running `kubectl` and the master nodes are able to reach AWS in
+order to get and validate tokens.
+
+
+### Installing
+
+The subordinate charm [aws-iam-authenticator](aws-iam-charm)
+and some relations are all that are required. This can be added with the
```
Comment (tvansteenburgh) [inline pages/k8s/aws-iam-auth.md]: s/This/These

File: pages/k8s/aws-iam-auth.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+following overlay([download it here][asset-aws-iam-overlay]):
+```yaml
+applications:
+  aws-iam:
+    charm: cs:~containers/aws-iam
+relations:
+  - ['aws-iam', 'kubernetes-master']
+  - ['aws-iam', 'easyrsa']
```
Comment (tvansteenburgh) [inline pages/k8s/aws-iam-auth.md]: Let's make sure we test with Vault also.

**Comment by tvansteenburgh** [comment]
Nicely done, thanks @hyperbolic2346

File: pages/k8s/aws-iam-auth.md [hunk 13]
```diff
@@ -0,0 +1,263 @@
...
+a Charmed Kubernetes cluster without RBAC enabled, any valid AWS IAM
+credential that can assume a role specified in the IAMIdentityMapping
+CRD will be able to run commands against the cluster. If RBAC is enabled,
+the user will have the permissions of the user defined in the
+IAMIdentityMapping CRD.
+
+### Enabling RBAC
+
+In order to get authorisation with AWS-IAM, you will need to use RBAC.
+Refer to the Charmed Kubernetes [RBAC documentation][k8s-rbac-docs] for complete options, but at
```
Comment (evilnick) [inline pages/k8s/aws-iam-auth.md]: there seems to be a missing word after 'at'.

Comment (tvansteenburgh) [reply pages/k8s/aws-iam-auth.md]: "at minimum you will need..."

**Pull request merged**